src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
       this.enableTouchActivity();
     }
   }
-});
 
-/**
- * Dispose of the component and all child components
- */
-Component.prototype.dispose = function(){
-  this.trigger({ type: 'dispose', 'bubbles': false });
+  // Temp for ES6 class transition, remove before 5.0
+  init() {
+    // console.log('init called on Component');
+    Component.apply(this, arguments);
+  }
 
-  // Dispose all children.
-  if (this.children_) {
-    for (var i = this.children_.length - 1; i >= 0; i--) {
-      if (this.children_[i].dispose) {
-        this.children_[i].dispose();
+  /**
+   * Dispose of the component and all child components
+   */
+  dispose() {
+    this.trigger({ type: 'dispose', 'bubbles': false });
+
+    // Dispose all children.
+    if (this.children_) {
+      for (var i = this.children_.length - 1; i >= 0; i--) {
+        if (this.children_[i].dispose) {
+          this.children_[i].dispose();
+        }
       }
     }
-  }
 
-  // Delete child references
-  this.children_ = null;
-  this.childIndex_ = null;
-  this.childNameIndex_ = null;
+    // Delete child references
+    this.children_ = null;
+    this.childIndex_ = null;
+    this.childNameIndex_ = null;
 
-  // Remove all event listeners.
-  this.off();
-
-  // Remove element from DOM
-  if (this.el_.parentNode) {
-    this.el_.parentNode.removeChild(this.el_);
-  }
+    // Remove all event listeners.
+    this.off();
 
-  Lib.removeData(this.el_);
-  this.el_ = null;
-};
+    // Remove element from DOM
+    if (this.el_.parentNode) {
```

Comment:
Should we use the `#el()` method instead of the "private" var when just reading this (throughout the module)?